packages/quantic/decisions/0001-testing-strategy.md

```diff
@@ -39,12 +39,15 @@ Key points for Playwright E2E tests:
 - **User Workflow Testing**: Focus on the essential paths users take to accomplish their goals with a given component, this for all the different Quantic use cases: Search, Insight, Case assist and Recomendations.
 - **Simulate User Actions**: Test realistic user actions, such as clicking buttons, entering data, and navigating through the UI.
 - **External System Interactions**: Verify interactions with external systems, like Coveo APIs and analytics, to ensure components behave as expected in a real-world environment. This also acts as a double test because it will also notify us by failing the test if the external API contract is broken too because we test for the expected request body but also the expected response as much as possible.
+- **Browser-specific behaviors**: Test browser-specific behaviors that would interact with our components.
 
 **Example Playwright Test Scenarios**:
 
 - A user uses a Quantic component by interacting with its multiple buttons and triggering Coveo analytics. We test that the expected analytics are sent, that the events are valid, and the response from the analytics API are correct.
 - A user navigates through a whole search workflow, interacting with multiple components. We test the expected Search API calls and responses, the analytics calls and response, and also the reactivity of the full search experience.
 - Verify the integration between the Quantic components and Salesforce. We test that modifying data through the Salesforce components has an impact and triggers the correct components reactions in a Quantic experience.
+- A user modifies the URL parameters. We assess the impact on our components.
+- The browser viewport size changes. We verify that our components adapt their display accordingly.
```

Comment:
This point is to be clarified as you can see in this PR, the viewport size logic can be tested with unit tests and sometimes is better tested in unit tests, I see the differentiation as the following: 
- The logic based on the viewport size that makes the component render its element can be be unit tested and is sometimes better unit tested like in this PR.
- The reactivity of the component to the resize events of the browser is something that can be E2E and is better E2E tested.

Bottom line is saying that only E2E tests are used to test such logics is untrue, as I'm noticing while evaluating how the QuanticTabBar component, the unit tests in this PR seem more useful.

Reply:
The VIEWPORT cannot be tested in unit tests, but the size of the container of an element can definitely be tested via unit tests.

That's why I suggested that since this was a bit more related to reactivity (aka how many tabs can be shown in a specific width) we should do those via playwright tests.

But I agree it's in a gray zone.